### PR TITLE
intel-mkl: add 2017.2

### PR DESCRIPTION
**Pull request by davydden**
p.s. I must admit I have some issues with this MKL, namely: 

> Name "ACTIVATION_LICENSE_FILE" is not valid

### CODE REVIEW

**Comment by adamjstewart** [comment]
Do you know if there are download URLs avaliable online like there are for intel-parallel-studio?

**Comment by davydden** [comment]
> Do you know if there are download URLs avaliable online like there are for intel-parallel-studio?

unfortunately, i don't.

**Comment by adamjstewart** [comment]
Found them:

| version | URL |
--- | ---
2017.2.174 | http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/11306/l_mkl_2017.2.174.tgz
2017.0.098 | http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/9662/l_mkl_2017.0.098.tgz
11.3.3.210 | http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/9068/l_mkl_11.3.3.210.tgz
11.3.2.181 | http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/8711/l_mkl_11.3.2.181.tgz

They appear to have the same checksums. Can you replace all of the URLs with these ones? The last 2 versions were out of order as well. Also, can you remove the note from the docstring that says it has to be manually downloaded and added to a mirror?

File: var/spack/repos/builtin/packages/intel-mkl/package.py [hunk 1]
```diff
@@ -29,26 +29,18 @@
...
-    """Intel Math Kernel Library.
-
-    Note: You will have to add the download file to a
-    mirror so that Spack can find it. For instructions on how to set up a
-    mirror, see http://spack.readthedocs.io/en/latest/mirrors.html.
-
-    To set the threading layer at run time set MKL_THREADING_LAYER
-    variable to one of the following values: INTEL (default), SEQUENTIAL, PGI.
-    To set interface layer at run time, use set the MKL_INTERFACE_LAYER
-    variable to LP64 (default) or ILP64.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/intel-mkl/package.py]: I think we should still keep this last paragraph. Just not the middle one.

Comment (davydden) [inline var/spack/repos/builtin/packages/intel-mkl/package.py]: no, we don't. This is outdated from times when we used `mkl_rt` in `intel-mkl`. Now everything is controlled via variants.

Comment (davydden) [inline var/spack/repos/builtin/packages/intel-mkl/package.py]: (we explicitly specify `mkl_integer` and `mkl_threading` libs to be used from MKL)

**Pull request merged**